install_config/install/advanced_install.adoc

```diff
@@ -681,6 +681,154 @@ openshift_examples_modify_imagestreams=true
+192.168.10.12 glusterfs_devices='[ "/dev/xvdc", "/dev/xvdd" ]'
+192.168.10.13 glusterfs_devices='[ "/dev/xvdc", "/dev/xvdd" ]'
+----
++
+The `glusterfs_devices` list of raw block devices will be completely managed as
+part of a GlusterFS cluster. There must be at least one device listed. Each
+device must be bare, with no partitions or LVM PVs.
+
+. Additional steps must be run manually after the cluster installation has
+completed. Continue with the rest of the preparation of your inventory file,
+then see xref:running-the-advanced-installation[Running the Advanced Installation] for the additional steps to complete the configuration of
+containerized GlusterFS persistent storage.
+
+[[advanced-install-configuring-openshift-container-registry]]
+=== Configuring the OpenShift Container Registry
+
+Additional configuration options are available at installation time for the
+xref:../../architecture/infrastructure_components/image_registry.adoc#integrated-openshift-registry[OpenShift Container Registry].
+
+[[advanced-install-glusterfs-backed-registry]]
+==== Configuring a GlusterFS-Backed Registry
+
+Similar to
+xref:advanced-install-containerized-glusterfs-persistent-storage[configuring containerized GlusterFS for persistent storage], a containerized
+GlusterFS-backed OpenShift Container Registry can be configured and deployed
+during the initial installation of the cluster to offer a more reliable
+container registry.
+
+If no registry storage options are used, the default {product-title} registry is
+ephermal and all data will be lost if the pod no longer exists. {product-title}
+also supports a single node NFS-backed registry, but this option lacks
+redundancy and reliability compared with the GlusterFS-backed option.
+
+To configure a containerized GlusterFS-backed registry:
+
+. Specific prerequisites and host preparation are required to run this deployment.
+See
+xref:../../install_config/install/prerequisites.adoc#prereq-containerized-glusterfs-considerations[Containerized GlusterFS Considerations] and ensure the steps have been satisfied.
+
+. In your inventory file, add `glusterfs_registry` to the `OSEv3:children` section
+to enable the `[glusterfs_registry]` group:
++
+----
+[OSEv3:children]
+masters
+nodes
+glusterfs_registry
+----
+
+. Add the following role variable in the `[OSEv3:vars]` section:
++
+----
+[OSEv3:vars]
+openshift_hosted_registry_storage_kind=glusterfs <1>
+----
+<1> Enables the GlusterFS-backed registry if the `glusterfs_registry` group name and
+the `[glusterfs_registry]` group exist.
+
+. The installer will deploy the OpenShift Container Registry pods and associated
+routers on nodes containing the `region=infra` label. Add this label on at least
+one node entry in the `[nodes`] section, otherwise the registry deployment will
+fail. For example:
++
+----
+[nodes]
+192.168.10.14 openshift_schedulable=True node=True openshift_node_labels="{'region': 'infra'}"
+----
+
```

Comment:
The whole of step 4 applies to ALL deployments of an integrated registry, not just one backed by GlusterFS, hence it should go above the GlusterFS section to prevent having to repeat it in any future subsections.